Source/Urho3D/Graphics/Direct3D11/D3D11ShaderVariation.cpp

```diff
@@ -172,23 +174,27 @@ void ShaderVariation::SetDefines(const ea::string& defines)
     defines_ = defines;
 }
 
-bool ShaderVariation::LoadByteCode(const ea::string& binaryShaderName)
+bool ShaderVariation::LoadByteCode(const FileIdentifier& binaryShaderName)
 {
-    ResourceCache* cache = owner_->GetSubsystem<ResourceCache>();
-    if (!cache->Exists(binaryShaderName))
+    const VirtualFileSystem* vfs = owner_->GetSubsystem<VirtualFileSystem>();
+    if (!vfs->Exists(binaryShaderName))
         return false;
 
     const FileSystem* fileSystem = owner_->GetSubsystem<FileSystem>();
     const unsigned sourceTimeStamp = owner_->GetTimeStamp();
     // If source code is loaded from a package, its timestamp will be zero. Else check that binary is not older
     // than source
-    if (sourceTimeStamp && fileSystem->GetLastModifiedTime(cache->GetResourceFileName(binaryShaderName)) < sourceTimeStamp)
-        return false;
+    if (sourceTimeStamp)
+    {
+        ea::string fullPath = vfs->GetAbsoluteNameFromIdentifier(binaryShaderName);
+        if (!fullPath.empty() && fileSystem->GetLastModifiedTime(fullPath) < sourceTimeStamp)
+            return false;
+    }
 
-    const AbstractFilePtr file = cache->GetFile(binaryShaderName);
+    const AbstractFilePtr file = vfs->OpenFile(binaryShaderName, FILE_READ);
     if (!file || file->ReadFileID() != "USHD")
     {
-        URHO3D_LOGERROR(binaryShaderName + " is not a valid shader bytecode file");
+        URHO3D_LOGERROR(binaryShaderName.fileName_ + " is not a valid shader bytecode file");
```

Comment:
Note for the future: if we are changing logging messages might as well convert them to using formats instead of keeping string concatenation.

> URHO3D_LOGERROR("{} is not a valid shader bytecode file", binaryShaderName.fileName_);